Upgraded dependencies versions, Go version, added support for schema …

lospejos opened the pull request:
Upgraded dependencies versions, Go version, added support for schema in tables definitions.

I made some effort to address issue I created earlier: 

https://github.com/Trendyol/go-pq-cdc/issues/9

Please check if my PR is full and, correct, I didn't have much time to test it. Thanks.

3n0ugh commented on README.md:
```diff
@@ -150,6 +150,7 @@ This setup ensures continuous data synchronization and minimal downtime in captu
 | `publication.tables`                    | []Table  |   yes    |    -    | Set tables which are tracked by data change capture                                                   | Define multiple tables as needed.                                                                                                                  |
 | `publication.tables[i].name`            |  string  |   yes    |    -    | Set the data change captured table name                                                               | Must be a valid table name in the specified database.                                                                                              |
 | `publication.tables[i].replicaIdentity` |  string  |   yes    |    -    | Set the data change captured table replica identity [`FULL`, `DEFAULT`]                               | **FULL:** Captures all columns when a row is updated or deleted. <br> **DEFAULT:** Captures only the primary key when a row is updated or deleted. |
+| `publication.tables[i].schema`          |  string  |    no    |    -    | Set the data change captured table schema name                                                        | Must be a valid table name in the specified database.                                                                                              |
```
Can we consider adding a default value for schema as public to maintain backward compatibility? This would ensure existing setups without a specified schema continue to function seamlessly.

3n0ugh commented on pq/publication/replica_identity.go:
```diff
@@ -66,10 +66,17 @@ func (c *Publication) GetReplicaIdentities(ctx context.Context) ([]Table, error)
 	}
 
-	query := "SELECT relname AS table_name, relreplident AS replica_identity FROM pg_class WHERE relname IN (" + strings.Join(tableNames, ", ") + ")"
+	query := "SELECT relname AS table_name, n.nspname AS schema_name, relreplident AS replica_identity FROM pg_class c JOIN pg_namespace n ON c.relnamespace = n.oid WHERE concat(n.nspname, '.', c.relname) IN (" + strings.Join(tableNames, ", ") + ")"
```
Can we consider using fmt.Sprintf here instead of concatenating strings directly? This could improve readability and reduce the risk of errors in constructing the PostgreSQL statement. What do you think?

3n0ugh reviewed (commented):
Can you also run the make audit command to check for lint and other potential issues? This will help ensure the code meets the project's standards.

3n0ugh commented on benchmark/go-pq-cdc-kafka/go.mod:
```diff
@@ -1,6 +1,6 @@
 module github.com/Trendyol/go-pq-cdc/benchmark/go-pq-cdc-kafka
 
-go 1.22.4
```
We are maintaining version and dependency upgrades separately. Could you please revert these changes from this feature?

lospejos commented:
@3n0ugh I've updated my PR:
1) Reverted Go and dependencies versions to original ones;
2) Added default value `public` for table schema in documentation and in the code;
3) Changed string concatenation to use `Sprintf` function instead of raw strings addition;

However, I could not execute `make audit` command since I have no `make` tool on Windows 10. I'm trying to figure out how to get it.

3n0ugh commented:
It seems the linter is causing [the test](https://github.com/Trendyol/go-pq-cdc/actions/runs/13013493458/job/36297140018?pr=10#step:4:205) to fail. On Windows 10, you can run the following commands to fix the issues:
Install the required tools:

```bash
go install golang.org/x/tools/go/analysis/passes/fieldalignment/cmd/fieldalignment@latest
go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
go install golang.org/x/vuln/cmd/govulncheck@latest
```
Ensure that `GOPATH/bin` is added to your `PATH` environment variable.

Run the commands:
```bash
fieldalignment -fix ./...
golangci-lint run -c .golangci.yml --timeout=5m -v --fix
go vet ./...
govulncheck ./...
```
This should resolve the linter issues and get the tests passing. Let me know if you run into any problems! 😊

lospejos commented:
> It seems the linter is causing [the test](https://github.com/Trendyol/go-pq-cdc/actions/runs/13013493458/job/36297140018?pr=10#step:4:205) to fail. On Windows 10, you can run the following commands to fix the issues: Install the required tools:
> 
> ```shell
> go install golang.org/x/tools/go/analysis/passes/fieldalignment/cmd/fieldalignment@latest
> go install github.com/golangci/golangci-lint/cmd/golangci-lint@latest
> go install golang.org/x/vuln/cmd/govulncheck@latest
> ```
> 
> Ensure that `GOPATH/bin` is added to your `PATH` environment variable.
> 
> Run the commands:
> 
> ```shell
> fieldalignment -fix ./...
> golangci-lint run -c .golangci.yml --timeout=5m -v --fix
> go vet ./...
> govulncheck ./...
> ```
> 
> This should resolve the linter issues and get the tests passing. Let me know if you run into any problems! 😊

Thanks for your help, I installed `fieldalignment`, `golangci-lint`, `govulncheck`, executed commands:

```
fieldalignment -fix ./...
golangci-lint run -c .golangci.yml --timeout=5m -v --fix
go vet ./...
govulncheck ./...
```
No errors was found, no source code changes to commit.

However, I see some errors in build/CI scripts, will inspect them.

lospejos commented:
> It looks like there’s an issue with the integration test step in the CI. You can run the test locally using:
> 
> ```shell
> make test/integration
> ```
> 
> This should help identify and resolve the problem.

As a mentioned earlier, I have no `make` installed. The binary in `integration_test` compiles without errors using:

```
> go mod tidy & go mod vendor & go build .
```

I'll try to find out the error cause and fix it.

lospejos commented:
@3n0ugh BTW when I run 

```
$ golangci-lint run -c .golangci.yml --timeout=5m -v --fix
```

on my `Linux Mint 22.1 Xia` with `Go 1.22.4` I see warning:
```
WARN The linter 'exportloopref' is deprecated (since v1.60.2) due to: Since Go1.22 (loopvar) this linter is no longer relevant. Replaced by copyloopvar.
```

lospejos commented:
@3n0ugh It seems that there is no more error:
```
[cannot call pointer method DSN on "github.com/Trendyol/go-pq-cdc/config".Config](https://github.com/Trendyol/go-pq-cdc/pull/10/commits/2d0f0e9b873ed90afd603d7ecef33c4f5711a520)
```

3n0ugh commented:
Everything looks good to me. Thank you for your effort and contribution! 
This pull request will be merged. 🎉
